src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -88,30 +101,75 @@
 from snowflake.connector import DictCursor, ProgrammingError
 from snowflake.connector.cursor import SnowflakeCursor
 
+ApplicationPackageChildrenTypes = (
+    StreamlitEntityModel | FunctionEntityModel | ProcedureEntityModel
+)
+
+
+class ApplicationPackageChildIdentifier(UpdatableModel):
+    schema_: Optional[str] = Field(
+        title="Child entity schema", alias="schema", default=None
+    )
+
+
+class EnsureUsableByField(UpdatableModel):
+    application_roles: Optional[Union[str, Set[str]]] = Field(
+        title="One or more application roles",
+        default=None,
+    )
+
+    @field_validator("application_roles", mode="before")
+    @classmethod
+    def ensure_app_roles_is_a_set(
+        cls, application_roles: Optional[Union[str, Set[str]]]
+    ) -> Optional[Union[str, Set[str]]]:
+        if isinstance(application_roles, str):
+            return set(application_roles)
+        return application_roles
+
+
+class ApplicationPackageChildField(UpdatableModel):
+    target: str = Field(title="The key of the entity to include in this package")
+    ensure_usable_by: Optional[EnsureUsableByField] = Field(
+        title="Use to automatically grant USAGE privilege on the child object",
```

Comment:
doesn't it also grant to the schema?
Maybe something like "Ensure child object is usable by the application role"? - basically if we're moving away from grant_usage syntax, we should probably update the title to match.

Reply:
Changed to:
> Automatically grant the required privileges on the child object and its schema